pkg/transform/mapper.go

```diff
@@ -109,6 +109,9 @@ func jqMapperConfigFunction(c *JQMapperConfig) (TransformationFunction, error) {
 	}
 
 	withEpochFunction := gojq.WithFunction("epoch", 0, 1, func(a1 any, a2 []any) any {
+		if a1 == nil {
+			return nil
+		}
 		return a1.(time.Time).UnixMilli()
```

Comment:
Come to think of it, nil values aren't our only case. If you call this function on a non-`time.Time` value, I think this will cause a panic and the whole app will crash. So I think we'll need to handle it specifically...

```
	data := map[string]interface{}{"empty": nil, "time": time.Now(), "string": "ceaeefa"}

	val, ok := data["string"].(time.Time)
	if !ok {
		fmt.Println("not a time")
	} else {
		fmt.Println(val.UnixMilli())
	}

```

Now, there's a question of what should the behaviour be if it's not a time value... nil should always result in nil. But for other types, we can:

- Always produce nil and process the data normally
- Create an error and produce an invalid

I think it should be the second option, but not very firmly entrenched in that point of view. Thoughts?

Reply:
Aaaaa you're right, we have a panic then....I don't know why but I assumed for some reason we'll have nice error and invalid message...

>I think it should be the second option

I agree